packages/rrweb/src/record/observer.ts

```diff
@@ -109,7 +109,14 @@ export function initMutationObserver(
   }
 
   const observer = new mutationObserverCtor(
-    callbackWrapper(mutationBuffer.processMutations.bind(mutationBuffer)),
+    callbackWrapper((mutations) => {
+      // If this callback returns `false`, we do not want to process the mutations
+      // This can be used to e.g. do a manual full snapshot when mutations become too large, or similar.
+      if (options.onMutation && options.onMutation(mutations) === false) {
```

Comment:
And the idea is that we then trigger this full snapshot from the Replay integration, right?

Reply:
Jup, we can control what to do from replay then!